### ISSUE TITLE
localization UI : make it more prominent with a snackbar and UI improvements

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#2731]
#### User story

Janice wants to read the site in Urdu. Or more to the point, she would like to, but she doesn't know it is possible. So she just makes do with English. 

#### Feature description

the current language selection UI is basically copied from Youtube. It's good and won't change majorly. But it has always been the intention to make it more prominent when we have more localizations done. That time is, I think, now.

How best to raise awareness?

Let's start minimal: just something to gently nudge a user, give them a bit of info, without affecting the overall UI.

The easiest way to do this is with a "toast" or "snackbar". A snackbar is a temporary popup that slides up from the bottom of the page (like a piece of toast from a toaster, hence they are sometimes called "toast"). 

https://m2.material.io/components/snackbars#anatomy

- A snackbar *does not block* the UI (unlike a modal)
- It contains one short bit of text and an optional action
- it appears for between four to ten seconds
- there is no close button, it disappears by itself
- it appears only once, not on subsequent visits.

Something like:

>You can change language on SuttaCentral.          <kbd> LEARN MORE </kbd>

Click <kbd> LEARN MORE </kbd>

It takes you to a static page which explains the language options. This is an enhanced version of the current "languages" page. 

https://suttacentral.net/languages?lang=en

It can also be linked from the top of the language selector:

>**Choose your language**
Where available, view translations and site UI in selected language. <kbd> LEARN MORE </kbd>

On the Languages page, it has something like:

>SuttaCentral hosts translations in over forty languages. By default the site is in English, but you can choose to read the site in your language, if translations exist. 
>
>Select your language from the list below. Change it at any time from this page, or from the three-dot menu at the top right of every page.
>
>Selecting your language will do two things.
>
>- The site interface will be in your language, if it has been translated.
>- Sutta translations in your language will be highlighted. Look for the "yellow bricks" in the navigation, and "follow the yellow brick road" to translations in your language. 

![yellow-brick-road](https://github.com/suttacentral/suttacentral/assets/6112010/01452cb3-8443-4703-996f-ce32d0fa3fed)

Okay, then let's adjust the list of languages to match the list in the menu a little better. 

Currently we have a simple list, with number of texts, and a link to a page with details per language.

>- Hebrew (90)

Let's make it more like:

<table>
<caption>Languages on SuttaCentral</caption>
<tr>
<th>Language</th>
<th>Select language</th>
<th>Number of texts</th>
<th>More details</th>
</tr>
<tr>
<td><kbd>PL</kbd>  Polish</td>
<td>  </td>
<td> 90 </td>
<td>↗</td>
</tr>
<tr>
<td><kbd>HE</kbd>  Hebrew</td>
<td> 🔘 </td>
<td> 90 </td>
<td>↗</td>
</tr>
</table>

- The radio button will select one and only one language. 
- "More details" is a link to the same page as is currently linked on the list, eg. https://suttacentral.net/languages/he?lang=en
- Users can also be given an option to select language on the "more details" page.

***

This also show up a weakness in the current menu implementation: there is no visual indication that a language has been selected.

We should have radio buttons on the right of each item in the menu, and show which one is selected. 

It is important that the selection via the languages page and via the menu be kept in sync!

**Comment by thesunshade** [comment]
I was actually just thinking about all this yesterday. Here are some thoughts…

>Sutta translations in your language will be highlighted. Look for the "yellow bricks" in the navigation, and "follow the yellow brick road" to translations in your language.

This is such an English language culturally specific reference that I believe it is more confusing than helpful. I 100% know that the Sinhala speaking monks I work with would have no idea what it is talking about. Since the whole point of this effort is to help *non* English speakers, I believe that it should just be explicit in plain language. 

>Sutta translations in your language will be highlighted. Look for the gold box in the top right corner of the navigation with the number of texts translated in your language, and follow the navigation to translations in your language.

Currently selecting a language from the three dot menu closes the whole menu. However when you go back to the menu again, it's *still* in the language list. I know that you have a principle of "don't do things for the user." However I'm constantly finding myself opening the three dot menu and finding the language selector open. Why? I've already chosen my language. If you aren't going to reset the menu to the main menu, then you shouldn't close it when I select a language.

Do you really think it's good to duplicate the language selecting system? Why not just tell people to choose it from the three dot menu.

I'm also wondering how that table will end up looking on mobile.

Is it possible to select the browser language and offer that as a "click to change to X" in the snack bar? I don't think it should change it without the user's consent, but it might be nice to save them some time. Might be too complicated, though.

**Comment by thesunshade** [comment]
Ven. @sabbamitta and I have been chatting about this on gitter.

## New new-user story

Helga has her browser language set to Deutch. When she opens SuttaCentral for the first time it should honour that prefrence and be in Deutch until she changes her settings.

The "language onboarding" process in the original issue should be done in Deutch.  Even though we can be fairly certain that Helga will want to use the site in Deutch, it's also possible that she is bilingual with French (her mother is from Switzerland) and so she may want to know the impact of having Deutch as her site language.

**Comment by sabbamitta** [comment]
> Even though we can be fairly certain that Helga will want to use the site in Deutch, it's also possible that she is bilingual with French (her mother is from Switzerland) and so she may want to know the impact of having Deutch as her site language.

I think whichever language a user has set in their browser, we can't know whether or not this is their native language, or their most fluent language, or the language they are planning to use over the next 100 years---but what we can conclude is that it is a language of which they have at least some basic understanding.

So if Sri Lankan monks like to read the internet in English and set their browser language accordingly, that means they *understand* it.

Whether Helga has set her browser to Deutsch or to Français is up to her; she probably won't select a language she doesn't understand at all.

So I think the browser language is a good guide to make sure the user has at least some basic understanding. And in that language they would understand instructions on how to change the language settings.

Many websites open in the browser language.

**Comment by firepick1** [comment]
Voice opens in the browser language. Voice also has a minimalist first-time-user  tutorial that directs users to Settings for whatever they wish to customize.  In the settings, users can choose:

* primary translation language
* web language
* reference language for comparing suttas

In Europe there are a lot of multi-lingual people, so this flexibility matters. I sometimes use three different languages for those three settings.  Because of this, we've shied away from crafting anything special for first time users regarding language settings. They can usually deal with the browser language and can optimize their experience later via Settings.

 
https://sc-voice.net

**Comment by thesunshade** [comment]
I agree with everything you say @sabbamitta. My point was just that I think everyone should experience the language onboarding.

I could only find [stats for FireFox](https://data.firefox.com/dashboard/usage-behavior), but I'm shocked to see that only ~40% of users have English as their interface language, if I'm reading it correctly. As an English user who has only ever installed a browser while in an English speaking country, I'm totally ignorant. I'm guessing that for non-English users there may be points in the setup process that trigger having a non-English language setting.

**Comment by sabbamitta** [comment]
>I'm guessing that for non-English users there may be points in the setup process that trigger having a non-English language setting.

Well unless I actively take action, any browser that I install on my computer will automatically have language set to DE, as this is the language of my computer. Usually when you buy a computer in Germany, this is the default. You also find German keyboards in the shops by default.

**Comment by noeismet** [comment]
> So I think the browser language is a good guide to make sure the user has at least some basic understanding. And in that language they would understand instructions on how to change the language settings.
> 
> Many websites open in the browser language.

Yes, I second that. This would surely help.  I'd still try to put the language icon more evident, but having the site in the browser's language would be great.
